pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -472,11 +472,23 @@ async def sign_artifacts(self, release_name: str, client_type: str, release_info
         async with await lock_manager.lock(lock_name):
             async with AsyncSignatory(uri, cert_file, key_file, sig_keyname=sig_keyname) as signatory:
                 tasks = []
+                json_digests = []
                 for release_info in release_infos.values():
+                    version = release_info["metadata"]["version"]
                     pullspec = release_info["image"]
                     digest = release_info["digest"]
+                    json_digests.append((version, pullspec, digest))
+                    # if this payload is a manifest list, iterate through each manifest
+                    manifests = release_info.get("manifests", [])
+                    for manifest in manifests:
+                        if manifest["platform"]["os"] != "linux":
+                            raise ValueError("Unsupported OS %s in manifest list %s", manifest["platform"]["os"], release_info["image"])
```

Comment:
Out of curiosity: Why is this?

Reply:
Maybe worthy of a comment

Reply:
This is actually a copy-pasta from https://github.com/openshift-eng/aos-cd-jobs/blob/b0063c829947db4c5950035ee993be603ce4e9c2/pyartcd/pyartcd/pipelines/promote.py#L369.
For now the manifest always has `platform.os==linux`. The check is not necessary but kinda defensive programming.